lib/pos_epoch_complete_hook.go

```diff
@@ -163,6 +162,43 @@ func (bav *UtxoView) runEpochCompleteEpochRollover(epochNumber uint64, blockHeig
 		FinalBlockHeight:                nextEpochFinalBlockHeight,
 		CreatedAtBlockTimestampNanoSecs: blockTimestampNanoSecs,
 	}
+	return nextEpochEntry, nil
+}
+
+func (bav *UtxoView) computePrevEpochEntry(epochNumber uint64, blockHeight uint64, view uint64, blockTimestampNanoSecs uint64) (*EpochEntry, error) {
+	if epochNumber == 0 {
+		return nil, errors.New("computePrevEpochEntry: epochNumber is 0, this should never happen")
+	}
+	snapshotAtEpochNumber, err := bav.ComputeSnapshotEpochNumberForEpoch(epochNumber - 1)
+	if err != nil {
+		return nil, errors.Wrapf(err, "computePrevEpochEntry: problem computing snapshot epoch number: ")
+	}
+	snapshotGlobalParamsentry, err := bav.GetSnapshotGlobalParamsEntryByEpochNumber(snapshotAtEpochNumber)
+	if err != nil {
+		return nil, errors.Wrapf(err, "computePrevEpochEntry: problem retrieving snapshot global params entry: ")
+	}
+
+	// Calculate the PrevEpoch's InitialBlockHeight.
+	prevEpochInitialBlockHeight, err := SafeUint64().Sub(blockHeight, snapshotGlobalParamsentry.EpochDurationNumBlocks+1)
```

Comment:
No need for the extra +1. For example, if each epoch is 3 blocks long, and the current epoch starts at block 4, then we know that the previous epoch contains blocks [1, 2, 3]. It's initial block height is 4 -3 = 1
```suggestion
	prevEpochInitialBlockHeight, err := SafeUint64().Sub(blockHeight, snapshotGlobalParamsentry.EpochDurationNumBlocks)
```